lib/scala/project-manager/src/main/scala/org/enso/projectmanager/infrastructure/languageserver/LanguageServerExecutor.scala

```diff
@@ -0,0 +1,158 @@
+package org.enso.projectmanager.infrastructure.languageserver
+
+import java.lang.ProcessBuilder.Redirect
+import java.util.concurrent.Executors
+
+import akka.actor.ActorRef
+import org.apache.commons.lang3.concurrent.BasicThreadFactory
+import org.enso.loggingservice.LogLevel
+import org.enso.projectmanager.infrastructure.languageserver.LanguageServerExecutor.SpawnResult
+import org.enso.projectmanager.service.versionmanagement.RuntimeVersionManagerMixin
+import org.enso.projectmanager.versionmanagement.DistributionConfiguration
+import org.enso.runtimeversionmanager.runner.{LanguageServerOptions, Runner}
+
+import scala.concurrent.{
+  ExecutionContext,
+  ExecutionContextExecutor,
+  Future,
+  Promise
+}
+
+/** An executor that allows to start the language server process.
+  *
+  * @param descriptor options related to this language server instance
+  * @param progressTracker reference to an actor that should be notifed of any
+  *                        locks
+  * @param rpcPort port to use for the RPC channel
+  * @param dataPort port to use for the binary channel
+  */
+class LanguageServerExecutor(
```

Comment:
The LanguageServerExecutor should act in a similar fashion as the real executor service. It should be a singleton. I should have a spawn method that takes as a parameter LS descriptor and LS lifecycle listener. This function should return handle or PID representing LS instance. The handle should be used to kill/stop the LS by the executor. Please redesign it as a service.

Reply:
But that is almost exactly what is done here, isn't it? The object below has the `spawn` function that acts as you described.

Do you mean that I should just make this class private as it is an implementation detail? Or do you mean that instead of it returning a `Future[Int]` I should provide it with a callback `onExited: Int => Unit`?

>  Please redesign it as a service.
Or do you mean that it should be an Actor? I'm not sure if I understood.

Reply:
I have reorganised this as we discussed, please have a look if it is fine now.